tests/cli/urls_test.py

```diff
@@ -30,6 +30,8 @@ def test_run(capsys: CaptureFixture) -> None:
 http://localhost:8000/metrics/{name}/: Return a metric by name.
 http://localhost:8000/metrics/{name}/data/: Return data for a metric.
 http://localhost:8000/metrics/{name}/sql/: Return SQL for a metric.
+http://localhost:8000/sql/data/: Return data for a DJ Query.
```

Comment:
Pls disregard the location of this comment, its just easy to see here. These new paths are confusing next to the existing `/queries/` path. How about ?
```
/queries/data/: Return data for a DJ Query.
/queries/sql/: Return SQL for a DJ Query.
```
Or did I miss the fact that these are special queries and you made a separate path on purpose?

Reply:
Agreed the current paths are confusing but I also feel that `"/queries/sql/"` will be confusing to new-comers who don't fully understand giving SQL to get different SQL. Since we'll need an endpoint for validating a query without running it, maybe we can offer that as a way to get the built sql? Then we wouldn't need to specify `data` in the endpoint.
```
POST /query/: Return data for a DJ Query
POST /query/validate/: Validate a DJ Query
```
The second one would also return the built SQL. Thoughts?

Reply:
Also I noticed the `GET "/queries/{query_id}/"` endpoint which should still work as expected. For consistency with my suggestions above, we should rename that to `GET "/query/{query_id}/"` (if we go with my suggestions, that is 
😄 ).

Reply:
I like your suggestion @samredai . Especially since I am not a fan of the REST paths starting with a plural term. I am not familiar with the REST standards so I won't push but imo this:
```
.../thing/{thing_id}
.../thing/list
```
is more intuitive than:
```
.../things/{thing_id}
.../things/
```

Reply:
@samredai so is this correct?

1. `/query`: the user sends a str query to get data
2. `/query/validate`: the user sends a str query to see if it is valid and gets back the query we would run
3. `/query/{query_id}`: I don't actually know what this does haha

and so I should make the `/sql` routes I have right now conform to 1, 2?

Reply:
Isn't `/query/{query_id}` more for tracking the results of a specific query? It would only make sense after someone has actually kicked off a query run. One potential flow:
* `/query` -> user sends query str, DJ returns a reference to a `query_id` that identifies the query run
* `/query/validate` -> users sends query str to see if it's valid
* `/query/{query_id}` -> returns the status of the query run
* `/query/{query_id}/data` -> returns the query data

Reply:
@shangyian that looks very sensible to me!

Reply:
@CircArgs yep that's right. And @shangyian good catch, it makes sense to me to split out the data retrieval part to a `/query/{query_id}/data/` endpoint. Right now it just shows up in the status response when the status is "COMPLETED" but that's not ideal. I like the explicit follow-up call to get the actual data.